### PR TITLE
Tracking event collection & API

### PR DESCRIPTION
**Pull request by mutaphysis**
Added a simple event reporting API that allows sending
tracking events to the accessibility apps backend.

 Required url parameters:
  - type: Event type
  - timestamp: UTC epoch timestamp in seconds
  - userUUID: client generated uuid of a user

 Example request:

  ```
  curl -X POST
       'localhost:4000/tracking-events/report?type=AttributeChanged&appToken=ec196bd1cf8c265e09d62fbd4654b57e&timestamp=1554973889&userUUID=1234'
       -H 'Accept: application/json'
       -H 'Content-Type: application/json'
       -d '{ "organizationId": "iaS2FtEqcps6F7nTh", "placeInfoId": "12455" }'
  ```

### CODE REVIEW

File: imports/both/api/TrackingEvents/TrackingEventSchema.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  timestamp: Date,
+  userUUID: String,
+  hashedIp: String,
+  userLocation: {
+    type: Array,
+    optional: true,
+    minCount: 2,
+    maxCount: 2,
+  },
+  'userLocation.$': {
```
Comment (opyh) [inline imports/both/api/TrackingEvents/TrackingEventSchema.ts]: Seems not to match the type, I think this should be a `PointGeometry`?

File: imports/both/api/TrackingEvents/TrackingEvent.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import IMappingEvent from '../MappingEvents/IMappingEvent';
+
+type BaseTrackingEvent = {
+  type: string,
+  timestamp: Date,
+  userUUID: string,
+  hashedIp: string,
+  userLocation?: {
```
Comment (opyh) [inline imports/both/api/TrackingEvents/TrackingEvent.ts]: How about `PointGeometry` from the `geojson` package?

File: imports/both/api/TrackingEvents/TrackingEventSchema.ts (hunk 1 above)
Comment (mutaphysis) [reply imports/both/api/TrackingEvents/TrackingEventSchema.ts]: Good catch!

Comment (mutaphysis) [reply imports/both/api/TrackingEvents/TrackingEventSchema.ts]: Will change, need to adapt the client as well. Good catch

**Comment by mutaphysis** [comment]
@opyh I added a bunch of changes for incrementing statistics into collections, can you take another look?

**Pull request merged**